### PR TITLE
ansible: initial support for podman

### LINKED ISSUE #1517: Integration of Podman
RHEL systems are moving from Docker to CRI-O container tools, like Podman, Buildah, etc; instead of the Docker engine.
This issue will track the work done to integrate these tools into Ansible playbooks that deploy Skydive.

### PR DESCRIPTION
**Pull request by EmilienM**
Background:
libpod aka Podman is library for running OCI-based containers in Pods.
It is likely going to replace Docker in newer versions of RHEL systems,
which means Docker won't be supported in the future.

The CLI is supposed to work like it did with Docker (with some changes);
so this patch is pretty basic. It adds skydive_container_cli variable
(default to docker for backward compatibiility) and replace the
hardcoded "docker" commands by using this variable instead.

Some adjustments were made:
- podman doesn't need docker socket access.
- podman doesn't support --restart natively, we need a systemd service.
- podman doesn't require to have docker started.

Future iterations will be made to support proper SystemD/Podman
integration like it was done in TripleO for other containers:
http://my1.fr/blog/openstack-containerization-with-podman-part-2-operations/

Closes #1517
Signed-off-by: Emilien Macchi <emilien@redhat.com>

skip skydive-go-fmt
skip skydive-functional-tests-backend-elasticsearch
skip skydive-functional-tests-backend-orientdb
skip skydive-scale-tests
skip skydive-cdd-overview-tests
skip skydive-unit-tests
skip skydive-compile-tests
skip skydive-k8s-tests
skip skydive-ppc64le-tests
skip skydive-functional-hw-tests

### CODE REVIEW

**Comment by EmilienM** [comment]
run vagrant-tests

**Comment by safchain** [comment]
ok to test

**Comment by safchain** [comment]
@EmilienM I just invited you, once you accepted (please do :) we will be able to trigger CI job

**Comment by lebauce** [comment]
run skydive-vagrant-tests

**Comment by EmilienM** [comment]
run skydive-vagrant-tests

**Comment by lebauce** [comment]
run vagrant-tests

**Comment by lebauce** [comment]
run vagrant-tests

**Comment by EmilienM** [comment]
run vagrant-tests

**Comment by EmilienM** [comment]
run vagrant-tests

**Comment by safchain** [comment]
run vagrant-tests

**Comment by safchain** [comment]
@EmilienM the last run reported a recursive LOOP error in the roles, I didn't check yet.

**Comment by EmilienM** [comment]
@safchain thanks for looking into it.

I looked at the logs and saw:

```The error appears to have been in '/var/lib/jenkins/workspace/skydive-vagrant-tests/src/github.com/skydive-project/skydive/contrib/vagrant/provisioning/playbook.yml.sample': line 22, column 5, but may```

My patch doesn't touch playbook.yml.sample, I'll take a look later today.

File: contrib/ansible/roles/skydive_analyzer/tasks/main.yml [hunk 1]
```diff
@@ -33,7 +33,7 @@
   tags:
     - keystone
 
-- include: docker.yml
+- include: main.yml
```
Comment (EmilienM) [inline contrib/ansible/roles/skydive_analyzer/tasks/main.yml]: the issue is here, I'll fix it

**Comment by EmilienM** [comment]
run vagrant-tests

**Comment by EmilienM** [comment]
run vagrant-tests

**Comment by EmilienM** [comment]
run vagrant-tests

**Comment by EmilienM** [comment]
run vagrant-tests

**Comment by EmilienM** [comment]
@safchain @lebauce I'll need help on that one, for the logs. This time it fails on actual skydive tests. Thanks

**Comment by safchain** [comment]
rerun skydive-vagrant-tests

**Comment by EmilienM** [comment]
Feb 20 10:05:29 agent1 skydive[8299]: 2019-02-20T10:05:29.566Z        ERROR        lldp/lldp.go:329 (*Probe).handleNode        agent1: Failed to add multicast address: invalid argument

is it expected?

**Comment by safchain** [comment]
@EmilienM sorry for the delay, the CI machines were off due to office relocation. For the errors we are working on fixing them.

**Comment by lebauce** [comment]
run skydive-vagrant-tests

**Comment by lebauce** [comment]
run skydive-vagrant-tests

**Comment by lebauce** [comment]
run skydive-vagrant-tests

**Comment by lebauce** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
rerun skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by safchain** [comment]
run skydive-vagrant-tests

**Comment by lebauce** [comment]
run skydive-vagrant-tests

**Pull request merged**